### PR TITLE
Remove non-threadsafe stacks from tree struct.

### LINKED ISSUE #459: Remove non-threadsafe stacks from tsk_tree_t
tsk_tree_t current has two stack variables which are used in one or two functions (making them non-threadsafe). Remove these and malloc the stacks each time instead.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Closes #459

@daniel-goldstein, can you have a quick check through this please?

### CODE REVIEW

File: c/tskit/convert.c [hunk 1]
```diff
@@ -46,6 +46,7 @@ typedef struct {
     size_t precision;
     tsk_flags_t options;
     char *newick;
+    tsk_id_t *traversal_stack;
```
Comment (daniel-goldstein) [inline c/tskit/convert.c:49]: Is there any reason to be concerned about thread safety for the newick converter?

Comment (jeromekelleher) [reply c/tskit/convert.c:49]: No, but we want to get rid of the stacks that are in the ``tsk_tree_t`` structure, as that was a Bad Idea in the first place.

Comment (daniel-goldstein) [reply c/tskit/convert.c:49]: Looks good then!

**Comment by jeromekelleher** [comment]
Aha, testing on Windows to the rescue! Turns out I had made the stacks 1 too small in the MRCA case  (which was correct before, but I had dismissed the + 1 as pointless), and this problem only appeared in the Python testing on Windows.

So, it's worth testing on Windows even if you don't really expect many people to use the software on Windows, just for a bit of platform diversity.

**Comment by daniel-goldstein** [comment]
Interesting! I also figured that the + 1 was pointless. Why is it necessary?

**Comment by daniel-goldstein** [comment]
Oh I see, the root node pushes its parent, null, onto the stack

**Comment by jeromekelleher** [comment]
> Oh I see, the root node pushes its parent, null, onto the stack. Would be good to have a test for the mrca of two isolated samples, that would expose the bug.

Excellent call, done.

**Pull request merged**